analysis/src/Files.ml

```diff
@@ -102,3 +102,9 @@ let classifySourceFile path =
   if Filename.check_suffix path ".res" && exists path then Res
   else if Filename.check_suffix path ".resi" && exists path then Resi
   else Other
+
+let writeTempFile ~prefix ~suffix contents =
```

Comment:
This was intended to be temporary as I figured out how we can modify the parser to accept string content rather than forcing you to supply a file it can read from. Will try and see if I can modify the parser to support that.

As for the question itself; no, I believe this can throw an exception that wouldn't be caught here. I'll fix that if I don't think modifying the parser to support raw content is feasible to fit into this diff.